src/nwb2bids/bids_models/_dataset_description.py

```diff
@@ -54,6 +54,10 @@ class DatasetDescription(BaseMetadataModel):
         description="The version of the BIDS standard that was used.",
         pattern=r"^\d+\.\d+(?:\.\d+)?$",
     )
+    HEDVersion: str | None = pydantic.Field(
+        description="The version of the HED standard that was used.",
+        pattern=r"^\d+\.\d+(?:\.\d+)?$",
```

Comment:
Maybe could be addressed in a followup, (and this seems strange to me tbh), but IIUC this can be a single item, or a list, and the list can contain [partnered libaray schemas](https://bids-specification.readthedocs.io/en/stable/appendices/hed.html#partnered-library-schemas)

So I think this should be `List[str] | str | None`, and the regex should be relaxed, since apparently these are valid:
 -  `"HEDVersion": "score_1.0.0"` 
 - `"HEDVersion": ["8.1.0", "sc:score_1.0.0"`